protocol/kafka_client/client.go

```diff
@@ -82,10 +84,19 @@ func (c *Client) Close() error {
 	return nil
 }
 
-func (c *Client) SendAndReceive(request []byte) (Response, error) {
+func (c *Client) SendAndReceive(request kafka_interface.RequestI, stageLogger *logger.Logger) (Response, error) {
+	header := request.GetHeader()
+	apiType := utils.APIKeyToName(header.ApiKey)
```

Comment:
```suggestion
	apiName := utils.APIKeyToName(header.ApiKey)
```

(consistent with the `APIKeyToName` terminology)